src/app/modules/submission/encrypt-submission/__tests__/encrypt-submission.controller.spec.ts

```diff
@@ -292,4 +314,219 @@ describe('encrypt-submission.controller', () => {
+      MockUserService.getPopulatedUserById.mockReturnValueOnce(
+        okAsync(MOCK_USER),
+      )
+      MockAuthService.getFormAfterPermissionChecks.mockReturnValueOnce(
+        okAsync(MOCK_FORM),
+      )
+      const expectedError = new ResponseModeError(
+        ResponseMode.Encrypt,
+        ResponseMode.Email,
+      )
+      MockEncryptSubService.checkFormIsEncryptMode.mockReturnValueOnce(
+        err(expectedError),
+      )
+
+      const mockRes = expressHandler.mockResponse()
+
+      // Act
+      await handleStreamEncryptedResponses(MOCK_REQ, mockRes, jest.fn())
+
+      // Assert
+      expect(mockRes.status).toHaveBeenCalledWith(400)
+      expect(mockRes.json).toHaveBeenCalledWith({
+        message: expectedError.message,
+      })
+      // Check cursor retrieval not called.
+      expect(MockEncryptSubService.getSubmissionCursor).not.toHaveBeenCalled()
+    })
+
+    it('should return 403 when user does not have read permissions for form', async () => {
+      // Arrange
+      MockUserService.getPopulatedUserById.mockReturnValueOnce(
+        okAsync(MOCK_USER),
+      )
+      const expectedError = new ForbiddenFormError('no access')
+      MockAuthService.getFormAfterPermissionChecks.mockReturnValueOnce(
+        errAsync(expectedError),
+      )
+      const mockRes = expressHandler.mockResponse()
+
+      // Act
+      await handleStreamEncryptedResponses(MOCK_REQ, mockRes, jest.fn())
+
+      // Assert
+      expect(mockRes.status).toHaveBeenCalledWith(403)
+      expect(mockRes.json).toHaveBeenCalledWith({
+        message: expectedError.message,
+      })
+      expect(
+        MockEncryptSubService.checkFormIsEncryptMode,
+      ).not.toHaveBeenCalled()
+      // Check cursor retrieval not called.
+      expect(MockEncryptSubService.getSubmissionCursor).not.toHaveBeenCalled()
+    })
+
+    it('should return 404 when form cannot be found', async () => {
+      // Arrange
+      MockUserService.getPopulatedUserById.mockReturnValueOnce(
+        okAsync(MOCK_USER),
+      )
+      const expectedError = new FormNotFoundError('not found')
+      MockAuthService.getFormAfterPermissionChecks.mockReturnValueOnce(
+        errAsync(expectedError),
+      )
+      const mockRes = expressHandler.mockResponse()
+
+      // Act
+      await handleStreamEncryptedResponses(MOCK_REQ, mockRes, jest.fn())
+
+      // Assert
+      expect(mockRes.status).toHaveBeenCalledWith(404)
+      expect(mockRes.json).toHaveBeenCalledWith({
+        message: expectedError.message,
+      })
+      expect(
+        MockEncryptSubService.checkFormIsEncryptMode,
+      ).not.toHaveBeenCalled()
+      // Check cursor retrieval not called.
+      expect(MockEncryptSubService.getSubmissionCursor).not.toHaveBeenCalled()
+    })
+
+    it('should return 410 when form is already archived', async () => {
+      // Arrange
+      MockUserService.getPopulatedUserById.mockReturnValueOnce(
+        okAsync(MOCK_USER),
+      )
```

Comment:
this is more of personal preference but I find it more readable to place repeated mocks in a `beforeEach`. if you need to override the mock in the `beforeEach` for specific tests, can call `mockReset()` first. but that's just me

Reply:
oki i listen to you

Reply:
actually nvm, i think having these steps in the test makes it clearer what is being mocked :P